src/jsonrpc.rs

```diff
@@ -275,11 +275,5 @@ mod tests {
 
         let unknown_method_with_id = json!({"jsonrpc":"2.0","method":"foo","id":1});
         let _: Incoming = serde_json::from_value(unknown_method_with_id).unwrap();
-
-        let missing_method = json!({"jsonrpc":"2.0"});
-        let _: Incoming = serde_json::from_value(missing_method).unwrap();
-
-        let missing_method_with_id = json!({"jsonrpc":"2.0","id":1});
-        let _: Incoming = serde_json::from_value(missing_method_with_id).unwrap();
```

Comment:
Any particular reason why these two test cases were removed? I believe both of these messages are considered invalid according to the JSON-RPC spec, but I think we should still check that they parse as `Invalid`, shouldn't we?

Reply:
I see now. With these changes, the server now responds with status code `-32700` "parse error" instead of `-32600` "invalid request". Not only is this a breaking change, it also happens to go counter to the JSON-RPC spec. The issue is that certain invalid server requests look similar in form to legitimate client responses, making it difficult to know which status code to return in which case. I wonder if there's any way to "massage" the `serde` parsing in such a way that this is not an issue? :thinking: